### PR TITLE
code refactor / fix edition of actors question

### LINKED ISSUE #1017: actor question edition : default value converted to IDs
### steps to reproduce

- create a new actors question, set one or several users as defautl value
- save the question
- edit the questin again

### observed 
the defautl values are IDs separated by coma

### expected
the defautl values souhd be the names of users, as they were typed at initial creation of the question

### PR DESCRIPTION
**Pull request by btry**
### Changes description

when an administrator edits an existing question of type actors with a default value, the user accounts are replaced by their ID in database. This fix shows the login of users instead of IDs

Code refactor
- a performance optimization
- remove deprecated calls
- create methods to factorize code related to field type classname computation / instanciation
- fix the defautl value display when editing a question

### Checklist

Please check if your PR fulfills the following specifications:

- [ ] Tests for the changes have been added
- [ ] Docs have been added/updated

### References

Closes #1017

### CODE REVIEW

File: inc/field.class.php [hunk 1]
```diff
@@ -65,13 +65,23 @@ public function prepareQuestionInputForSave($input) {
...
     * Prepares an answer value for output in a target object
-    * @param string|array $input the answer to format for a target (ticket or change)
+    * @param  string|array $input the answer to format for a target (ticket or change)
     * @return string
     */
    public function prepareQuestionInputForTarget($input) {
       return Toolbox::addslashes_deep($input);
    }
 
+   /**
```
Comment (cedric-anne) [inline inc/field.class.php]: You should document it on interface, not on implementation.

File: inc/fields.class.php [hunk 2]
```diff
@@ -329,4 +328,39 @@ public static function updateVisibility($currentValues) {
 
       return $questionToShow;
    }
+
+   /**
```
Comment (cedric-anne) [inline inc/fields.class.php]: Documentation was copied from `fieldTypeExists()` and is not suitable for this method.

File: inc/fields.class.php [hunk 3]
```diff
@@ -329,4 +328,39 @@ public static function updateVisibility($currentValues) {
...
+      return (class_exists(self::getFieldClassname($type)));
+   }
+
+   /**
+    * gets an instance of a field given its type
+    *
+    * @param string $type type of field to get
+    * @param PluginFormcreatorQuestion $question question representing the field
+    * @param array $data additional data
+    */
```
Comment (cedric-anne) [inline inc/fields.class.php]: Missing `@return null|PluginFormcreatorFieldInterface`

File: inc/fields.class.php [hunk 4]
```diff
@@ -329,4 +328,39 @@ public static function updateVisibility($currentValues) {
...
+   }
+
+   /**
+    * checks if a field type exists
+    *
+    * @param string $type type of field to test for existence
+    * @return boolean
+    */
+   public static function fieldTypeExists($type) {
+      return (class_exists(self::getFieldClassname($type)));
```
Comment (cedric-anne) [inline inc/fields.class.php]: Adding something like `&& is_a(self::getFieldClassname($type), PluginFormcreatorFieldInterface::class, true)` could make code more defensive.

Comment (btry) [reply inc/fields.class.php]: I would also say that 
```php
is_a(self::getFieldClassname($type), PluginFormcreatorFieldInterface::class, true)
``` 
is sufficient by itself to check that the class exists.

Comment (cedric-anne) [reply inc/fields.class.php]: Indeed.

**Pull request merged**